### PR TITLE
Bug 1833245: Validating duration fields on image registry types

### PR DESCRIPTION
**Pull request by ricardomaraschini**
Adding a Format tag to validate "metav1.Duration" types on Image
Registry types.

### CODE REVIEW

File: imageregistry/v1/types.go [hunk 1]
```diff
@@ -301,6 +302,7 @@ type ImageRegistryConfigRequestsLimits struct {
 	// maxWaitInQueue sets the maximum time a request can wait in the queue
 	// before being rejected.
 	// +optional
+	// +kubebuilder:validation:Format=duration
```
Comment (dmage) [inline imageregistry/v1/types.go:305]: @ricardomaraschini is this format documented somewhere?

**Comment by ricardomaraschini** [comment]
/hold

Even though I tested manually this PR I can't guarantee this format really works as expected. I am putting this on hold until I found what the code that validates this format looks like.

File: imageregistry/v1/types.go (hunk 1 above)
Comment (ricardomaraschini) [reply imageregistry/v1/types.go:305]: After some digging around I managed to find out where this validation is made[1]. 

[1] https://github.com/go-openapi/strfmt/blob/master/duration.go

**Comment by ricardomaraschini** [comment]
/unhold

I have found out where this validation is made.

File: imageregistry/v1/types.go (hunk 1 above)
Comment (ricardomaraschini) [reply imageregistry/v1/types.go:305]: Documented here https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/#jsonschemaprops-v1-apiextensions-k8s-io

**Comment by dmage** [comment]
@ricardomaraschini sent me the link: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/#jsonschemaprops-v1-apiextensions-k8s-io

/lgtm

**Pull request merged**